src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/IdentifiersShouldNotContainUnderscoresTests.cs

```diff
@@ -1060,6 +1060,32 @@ End Sub
 
         #endregion
 
+        #region "CodeFix Tests"
+        [Fact]
+        public async Task CA1707_ClassNameHasUnderscores_Rename()
```

Comment:
Can you add some more tests based on the checks that you perform in the fixer - for example empty length check after RemoveUnderscores, `SyntaxFacts.IsValidIdentifier` check, `LookupSymbols` check.

Reply:
Also it might be worth having one test for each of the supported symbol types:  `Namespace`, `NamedType`, `Method`, `Property`, `Field`, `Event`. The easiest might be to update all existing tests and add for each the expected code fix.

Reply:
Also the field case is actually interesting because I think that the code-fix should not result in an upper-case for the first character when the field was having a leading underscore (e.g. `_user` should not result in `User` but in `user`).

Reply:
Oh also for fields, there might be a question for the static fields, some conventions use the `s_` prefix and you probably don't want to rename with an upper-case `S` (e.g. `s_myList` should not give `sMyList`).

Reply:
@Evangelink, I think `_user` and `s_myList` cases shouldn't be even reported from the analyzer side (I haven't checked if they're currently reported or not).

Reply:
I confirm that `_user` and `s_user` cases don't report any diagnostic. I would still add the few tests suggested by @mavasani and also I would consider this part of my first comment:

> The easiest might be to update all existing tests and add for each the expected code fix.